mmc-google-analytics/src/test/unit/java/info/novatec/metricscollector/google/ResponseParserTest.java

```diff
@@ -0,0 +1,131 @@
+package info.novatec.metricscollector.google;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.api.services.analyticsreporting.v4.model.ColumnHeader;
+import com.google.api.services.analyticsreporting.v4.model.DateRangeValues;
+import com.google.api.services.analyticsreporting.v4.model.GetReportsResponse;
+import com.google.api.services.analyticsreporting.v4.model.MetricHeader;
+import com.google.api.services.analyticsreporting.v4.model.MetricHeaderEntry;
+import com.google.api.services.analyticsreporting.v4.model.Report;
+import com.google.api.services.analyticsreporting.v4.model.ReportData;
+import com.google.api.services.analyticsreporting.v4.model.ReportRow;
+import com.google.common.collect.Lists;
+
+
+/**
+ * Created by DBE on 22.09.2017.
+ */
+public class ResponseParserTest {
+
+    ResponseParser responseParser;
+    List<Metrics> metricsList;
+
+    @Before
+    public void init(){
+        responseParser = new ResponseParser();
+        metricsList = new ArrayList<>();
+    }
+
+    @Test
+    public void parseEmptyResponse() {
+        responseParser.parse(createGetReportsResponse(0), metricsList);
+        assertThat(metricsList.size()).isEqualTo(0);
```

Comment:
Recommended approach is to check is list is Empty -> assertThat(metricsList).isEmpty();